### PR TITLE
New package: mitmproxy-8.1.1

### LINKED ISSUE #4432: Package Request: mitmproxy
A swiss-army knife for debugging, testing, privacy measurements, and penetration testing. It can be used to intercept, inspect, modify and replay web traffic such as HTTP/1, HTTP/2, WebSockets, or any other SSL/TLS-protected protocols

---
Would be nice to have it in repos.
Thanks for considering :)

### PR DESCRIPTION
**Pull request by unspecd**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements): **YES**

Resolves #4432.
Depends on ~#34769, #34770, #34772, #34773, #34775, #34777, #34791~.

### CODE REVIEW

**Comment by paper42** [comment]
please put all the dependencies to this PR

**Comment by unspecd** [comment]
Should I close those PRs?

**Comment by unspecd** [comment]
One test failed on i686: `OverflowError: timestamp out of range for platform time_t`.
I do not think this is a bug, I guess there's an error in the test.

**Comment by paper42** [comment]
> Should I close those PRs?

yes

**Comment by unspecd** [comment]
https://github.com/mitmproxy/mitmproxy/issues/5030#issuecomment-1003725946

I'll skip that test for now.

File: srcpkgs/mitmproxy/template [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+checkdepends="${depends} python3-hypothesis python3-parver
+ python3-pytest-asyncio python3-pytest-cov python3-pytest-timeout python3-requests"
+short_desc="Interactive TLS-capable intercepting HTTP proxy"
+maintainer="Evgeny Ermakov <evgeny.v.ermakov@gmail.com>"
+license="MIT"
+homepage="https://mitmproxy.org"
+distfiles="https://github.com/mitmproxy/mitmproxy/archive/v${version}.tar.gz"
+checksum=8728d18c69053f0043acebcdabf46f2eeea51f0f0b60c528e1d356cf48ed2ca2
+
+_skip="(test_get_version)" # This test fails without a git repository
```
Comment (unspecd) [inline srcpkgs/mitmproxy/template:21]: https://github.com/mitmproxy/mitmproxy/issues/5057#issuecomment-1013706881

**Comment by unspecd** [comment]
Updated to 8.0.0.

**Comment by unspecd** [comment]
Updated to 8.1.0.

**Comment by unspecd** [comment]
Updated to 8.1.1.

File: srcpkgs/mitmproxy/template [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ python3-click python3-cryptography python3-h11 python3-h2 python3-hyperframe python3-kaitaistruct
+ python3-ldap3 python3-msgpack python3-openssl python3-parsing python3-passlib python3-protobuf
+ python3-publicsuffix2 python3-pyperclip python3-pyzstd python3-ruamel.yaml python3-sortedcontainers
+ python3-tornado python3-urwid python3-wsproto python3-zstandard"
+checkdepends="${depends} python3-hypothesis python3-parver
+ python3-pytest-asyncio python3-pytest-cov python3-pytest-timeout python3-requests"
+short_desc="Interactive TLS-capable intercepting HTTP proxy"
+maintainer="Evgeny Ermakov <evgeny.v.ermakov@gmail.com>"
+license="MIT"
+homepage="https://mitmproxy.org"
```
Comment (classabbyamp) [inline srcpkgs/mitmproxy/template:17]:
```suggestion
homepage="https://mitmproxy.org"
changelog="https://raw.githubusercontent.com/mitmproxy/mitmproxy/main/CHANGELOG.md"
```

File: srcpkgs/python3-Arpeggio/template [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+wrksrc="Arpeggio-${version}"
+build_style=python3-module
+hostmakedepends="python3-setuptools python3-wheel"
+depends="python3"
+checkdepends="python3-pytest"
+short_desc="Recursive descent parser with memoization based on PEG grammars"
+maintainer="Evgeny Ermakov <evgeny.v.ermakov@gmail.com>"
+license="MIT"
+homepage="https://github.com/textX/Arpeggio"
+distfiles="${homepage}/archive/${version}.tar.gz"
```
Comment (classabbyamp) [inline srcpkgs/python3-Arpeggio/template:14]: expand $homepage here

File: srcpkgs/python3-asgiref/template [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+wrksrc="asgiref-${version}"
+build_style=python3-module
+hostmakedepends="python3-setuptools"
+depends="python3"
+checkdepends="python3-pytest python3-pytest-asyncio"
+short_desc="ASGI base libraries"
+maintainer="Evgeny Ermakov <evgeny.v.ermakov@gmail.com>"
+license="BSD-3-Clause"
+homepage="https://github.com/django/asgiref"
+distfiles="${homepage}/archive/${version}.tar.gz"
```
Comment (classabbyamp) [inline srcpkgs/python3-asgiref/template:14]: expand $homepage here

File: srcpkgs/python3-h2/template [hunk 5]
```diff
@@ -1,17 +1,18 @@
...
 build_style=python3-module
 hostmakedepends="python3-setuptools"
-depends="python3 python3-hpack python3-hyperframe"
+depends="python3-hpack python3-hyperframe"
+checkdepends="${depends} python3-hypothesis python3-pytest"
 short_desc="HTTP/2 State-Machine based protocol implementation (Python3)"
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="MIT"
 homepage="https://github.com/python-hyper/h2"
 distfiles="${homepage}/archive/v${version}.tar.gz"
```
Comment (classabbyamp) [inline srcpkgs/python3-h2/template:14]: expand $homepage here

File: srcpkgs/python3-parver/template [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+wrksrc="parver-${version}"
+build_style=python3-module
+hostmakedepends="python3-setuptools"
+depends="python3-Arpeggio python3-attrs python3-six"
+checkdepends="${depends} python3-hypothesis python3-pretend python3-pytest"
+short_desc="Allows parsing and manipulation of PEP 440 version numbers"
+maintainer="Evgeny Ermakov <evgeny.v.ermakov@gmail.com>"
+license="MIT"
+homepage="https://github.com/RazerM/parver"
+distfiles="${homepage}/archive/${version}.tar.gz"
```
Comment (classabbyamp) [inline srcpkgs/python3-parver/template:14]: expand $homepage here

File: srcpkgs/python3-publicsuffix2/template [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+wrksrc="python-publicsuffix2-release-2.${version//./-}"
+build_style=python3-module
+hostmakedepends="python3-setuptools python3-pip python3-wheel"
+depends="python3"
+checkdepends="python3-requests"
+short_desc="Get a public suffix for a domain name using the Public Suffix List"
+maintainer="Evgeny Ermakov <evgeny.v.ermakov@gmail.com>"
+license="MIT, MPL-2.0"
+homepage="https://github.com/nexB/python-publicsuffix2"
+distfiles="${homepage}/archive/release-2.${version//./-}.tar.gz"
```
Comment (classabbyamp) [inline srcpkgs/python3-publicsuffix2/template:14]: expand $homepage here

File: srcpkgs/python3-zstandard/template [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+build_style=python3-module
+hostmakedepends="python3-setuptools"
+makedepends="python3-devel"
+depends="python3"
+checkdepends="python3-hypothesis"
+short_desc="Python bindings to the Zstandard (zstd) compression library"
+maintainer="Evgeny Ermakov <evgeny.v.ermakov@gmail.com>"
+license="BSD-3-Clause"
+homepage="https://github.com/indygreg/python-zstandard"
+distfiles="${homepage}/archive/${version}.tar.gz"
```
Comment (classabbyamp) [inline srcpkgs/python3-zstandard/template:15]: expand $homepage here

File: srcpkgs/python3-Arpeggio/template [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+revision=1
+wrksrc="Arpeggio-${version}"
+build_style=python3-module
+hostmakedepends="python3-setuptools python3-wheel"
+depends="python3"
+checkdepends="python3-pytest"
+short_desc="Recursive descent parser with memoization based on PEG grammars"
+maintainer="Evgeny Ermakov <evgeny.v.ermakov@gmail.com>"
+license="MIT"
+homepage="https://github.com/textX/Arpeggio"
```
Comment (classabbyamp) [inline srcpkgs/python3-Arpeggio/template:13]:
```suggestion
homepage="https://github.com/textX/Arpeggio"
changelog="https://raw.githubusercontent.com/textX/Arpeggio/master/CHANGELOG.md"
```

File: srcpkgs/python3-asgiref/template [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+revision=1
+wrksrc="asgiref-${version}"
+build_style=python3-module
+hostmakedepends="python3-setuptools"
+depends="python3"
+checkdepends="python3-pytest python3-pytest-asyncio"
+short_desc="ASGI base libraries"
+maintainer="Evgeny Ermakov <evgeny.v.ermakov@gmail.com>"
+license="BSD-3-Clause"
+homepage="https://github.com/django/asgiref"
```
Comment (classabbyamp) [inline srcpkgs/python3-asgiref/template:13]:
```suggestion
homepage="https://github.com/django/asgiref"
changelog="https://raw.githubusercontent.com/django/asgiref/main/CHANGELOG.txt"
```

File: srcpkgs/python3-h2/template [hunk 11]
```diff
@@ -1,17 +1,18 @@
...
 wrksrc="h2-${version}"
 build_style=python3-module
 hostmakedepends="python3-setuptools"
-depends="python3 python3-hpack python3-hyperframe"
+depends="python3-hpack python3-hyperframe"
+checkdepends="${depends} python3-hypothesis python3-pytest"
 short_desc="HTTP/2 State-Machine based protocol implementation (Python3)"
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="MIT"
 homepage="https://github.com/python-hyper/h2"
```
Comment (classabbyamp) [inline srcpkgs/python3-h2/template:13]:
```suggestion
homepage="https://github.com/python-hyper/h2"
changelog="https://raw.githubusercontent.com/python-hyper/h2/master/CHANGELOG.rst"
```

File: srcpkgs/python3-parver/template [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+revision=1
+wrksrc="parver-${version}"
+build_style=python3-module
+hostmakedepends="python3-setuptools"
+depends="python3-Arpeggio python3-attrs python3-six"
+checkdepends="${depends} python3-hypothesis python3-pretend python3-pytest"
+short_desc="Allows parsing and manipulation of PEP 440 version numbers"
+maintainer="Evgeny Ermakov <evgeny.v.ermakov@gmail.com>"
+license="MIT"
+homepage="https://github.com/RazerM/parver"
```
Comment (classabbyamp) [inline srcpkgs/python3-parver/template:13]:
```suggestion
homepage="https://github.com/RazerM/parver"
changelog="https://raw.githubusercontent.com/RazerM/parver/master/CHANGELOG.md"
```

File: srcpkgs/python3-publicsuffix2/template [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+revision=1
+wrksrc="python-publicsuffix2-release-2.${version//./-}"
+build_style=python3-module
+hostmakedepends="python3-setuptools python3-pip python3-wheel"
+depends="python3"
+checkdepends="python3-requests"
+short_desc="Get a public suffix for a domain name using the Public Suffix List"
+maintainer="Evgeny Ermakov <evgeny.v.ermakov@gmail.com>"
+license="MIT, MPL-2.0"
+homepage="https://github.com/nexB/python-publicsuffix2"
```
Comment (classabbyamp) [inline srcpkgs/python3-publicsuffix2/template:13]:
```suggestion
homepage="https://github.com/nexB/python-publicsuffix2"
changelog="https://raw.githubusercontent.com/nexB/python-publicsuffix2/develop/CHANGELOG.rst"
```

File: srcpkgs/python3-zstandard/template [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+wrksrc="python-zstandard-${version}"
+build_style=python3-module
+hostmakedepends="python3-setuptools"
+makedepends="python3-devel"
+depends="python3"
+checkdepends="python3-hypothesis"
+short_desc="Python bindings to the Zstandard (zstd) compression library"
+maintainer="Evgeny Ermakov <evgeny.v.ermakov@gmail.com>"
+license="BSD-3-Clause"
+homepage="https://github.com/indygreg/python-zstandard"
```
Comment (classabbyamp) [inline srcpkgs/python3-zstandard/template:14]:
```suggestion
homepage="https://github.com/indygreg/python-zstandard"
changelog="https://raw.githubusercontent.com/indygreg/python-zstandard/main/docs/news.rst"
```

**Review by classabbyamp — COMMENTED** [review]
fix these small things and I think this can be merged

File: srcpkgs/python3-kaitaistruct/template [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+build_style=python3-module
+hostmakedepends="python3-setuptools python3-wheel"
+depends="python3"
+short_desc="Kaitai Struct API for Python"
+maintainer="Evgeny Ermakov <evgeny.v.ermakov@gmail.com>"
+license="MIT"
+homepage="https://kaitai.io"
+distfiles="https://github.com/kaitai-io/kaitai_struct_python_runtime/archive/${version}.tar.gz"
+checksum=d21207d17603d87f263c8adcd45235ce22d154ba6faba0945e4f1a9d8d95e7e9
+make_check=no  # Package ships no tests
```
Comment (paper42) [inline srcpkgs/python3-kaitaistruct/template:15]: this shouldn't be an issue, does the build fail when no tests are ran?

File: srcpkgs/python3-publicsuffix2/template [hunk 16]
```diff
@@ -0,0 +1,19 @@
+# Template file for 'python3-publicsuffix2'
+pkgname=python3-publicsuffix2
+version=2019.12.21
+revision=1
+wrksrc="python-publicsuffix2-release-2.${version//./-}"
+build_style=python3-module
+hostmakedepends="python3-setuptools python3-pip python3-wheel"
```
Comment (paper42) [inline srcpkgs/python3-publicsuffix2/template:7]: why pip?

Comment (unspecd) [reply srcpkgs/python3-publicsuffix2/template]: https://github.com/void-linux/void-packages/runs/7171655428?check_suite_focus=true#step:7:1299

Comment (paper42) [reply srcpkgs/python3-publicsuffix2/template:7]:
```
Command '['/usr/bin/python3', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/tmp/tmp3mvkx0wo', '--quiet', 'requests>=2.7.0']' returned non-zero exit status 1.
```
That only means you are missing requests from dependencies. Adding pip to hostmakedepends will just hide this issue and download requests in their preferred version, but we want to run the tests on Void packages.

File: srcpkgs/python3-kaitaistruct/template (hunk 15 above)
Comment (unspecd) [reply srcpkgs/python3-kaitaistruct/template]: Honestly I don't even remember why I added this line.

Comment (unspecd) [reply srcpkgs/python3-kaitaistruct/template]: Fixed.

**Comment by unspecd** [comment]
@classabbyamp
Fixed.

**Comment by Duncaen** [comment]
Is it correct that mitm proxy needs two zstd bindings, and is there a reason python3-zstandard is not using the system library?

**Comment by unspecd** [comment]
I have created PRs to address your remarks:
 - #38181
 - #38182

**Pull request merged**